lib/src/git.rs

```diff
@@ -1420,12 +1417,7 @@ fn push_refs(
         push_options.remote_callbacks(callbacks);
         remote
             .push(refspecs, Some(&mut push_options))
-            .map_err(|err| match (err.class(), err.code()) {
-                (git2::ErrorClass::Reference, git2::ErrorCode::NotFastForward) => {
-                    GitPushError::NotFastForward
-                }
-                _ => GitPushError::InternalGitError(err),
-            })
+            .map_err(GitPushError::InternalGitError)
```

Comment:
nit: `.map_err(..)` can be removed at all. (handled later by `?`)

Reply:
Nice!